chromium_src/chrome/browser/ui/views/profiles/avatar_toolbar_button_state_manager.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "chrome/browser/ui/views/profiles/avatar_toolbar_button_state_manager.h"
+
+#include "brave/app/vector_icons/vector_icons.h"
+#include "brave/components/vector_icons/vector_icons.h"
+#include "chrome/browser/profiles/profile.h"
+
+namespace {
+
+// Creates a Brave-specific state provider for the avatar toolbar button.
+// Please see comments on IsRegularProfile() override for more details.
+std::unique_ptr<StateProvider> CreateBraveStateProvider(
+    Profile* profile,
+    StateObserver* state_observer);
+
+}  // namespace
+
+// Overrides profile->IsRegularProfile() in CreateStatesAndListeners() to
+// customize state provider for Tor, Incognito, and Guest profiles.
+// StateProvider provides data to visualize the avatar toolbar button's state
+// according to various situations. But Brave has specific requirements for
+// Tor, Incognito, and Guest profiles to show different icons for these. So
+// in these cases, we use Brave-specific state providers and do not add any
+// other state providers.
+#define IsRegularProfile()                                                 \
+  IsTor() || profile->IsIncognitoProfile() || profile->IsGuestSession()) { \
+    states_[ButtonState::kNormal] =                                        \
```

Comment:
re comment below, if using kNormal is correct here (seems odd?) it needs some explanation

Reply:
cc: @sangwoo108

Reply:
Answered here - https://github.com/brave/brave-core/pull/29767#discussion_r2274824193